starboard/nplb/player_write_sample_test.cc

```diff
@@ -31,95 +33,114 @@ typedef SbPlayerTestFixture::VideoSamples VideoSamples;
 class SbPlayerWriteSampleTest
     : public ::testing::TestWithParam<SbPlayerTestConfig> {
  protected:
-  SbPlayerWriteSampleTest();
-
-  void SetUp() override;
-  void TearDown() override;
-
-  SbMediaType test_media_type_;
-  std::unique_ptr<SbPlayerTestFixture> player_fixture_;
-};
-
-SbPlayerWriteSampleTest::SbPlayerWriteSampleTest() {
-  auto config = GetParam();
-  const char* audio_filename = std::get<0>(config);
-  const char* video_filename = std::get<1>(config);
-  SbPlayerOutputMode output_mode = std::get<2>(config);
-
-  if (audio_filename) {
-    SB_DCHECK(!video_filename);
-    test_media_type_ = kSbMediaTypeAudio;
-  } else {
-    SB_DCHECK(video_filename);
-    test_media_type_ = kSbMediaTypeVideo;
+  SbPlayerWriteSampleTest() {
+    const SbPlayerTestConfig& config = GetParam();
+    const char* audio_filename = std::get<0>(config);
+    const char* video_filename = std::get<1>(config);
+    const SbPlayerOutputMode output_mode = std::get<2>(config);
+    const char* key_system = std::get<3>(config);
+
+    SB_LOG(INFO) << "Initialize SbPlayerWriteSampleTest with audio file '"
+                 << (audio_filename ? audio_filename : "no audio")
+                 << "', video file '"
+                 << (video_filename ? video_filename : "no video")
+                 << "', key system '"
+                 << (key_system ? key_system : "no key system")
+                 << "', and output mode '"
+                 << (output_mode == kSbPlayerOutputModeDecodeToTexture
+                         ? "Decode To Texture"
+                         : "Punchout")
+                 << "'.";
   }
-
-  SB_LOG(INFO) << FormatString(
-      "Initialize SbPlayerWriteSampleTest with dmp file '%s' and with output "
-      "mode '%s'.",
-      audio_filename ? audio_filename : video_filename,
-      output_mode == kSbPlayerOutputModeDecodeToTexture ? "Decode To Texture"
-                                                        : "Punchout");
-}
-
-void SbPlayerWriteSampleTest::SetUp() {
-  ASSERT_NO_FATAL_FAILURE(
-      player_fixture_.reset(new SbPlayerTestFixture(GetParam())));
-}
-
-void SbPlayerWriteSampleTest::TearDown() {
-  ASSERT_NO_FATAL_FAILURE(player_fixture_.reset());
-}
+};
 
 TEST_P(SbPlayerWriteSampleTest, SeekAndDestroy) {
-  ASSERT_NO_FATAL_FAILURE(player_fixture_->Seek(kSbTimeSecond));
+  SbPlayerTestFixture player_fixture(GetParam());
```

Comment:
The two lines are used in all tests, can you elaborate why we make the change?

Reply:
I think this style brings more flexibility when writing test case, like for multiple player tests, and tests like create player-> destroy player -> create another player -> destroy player.

Reply:
`EXPECT_FALSE(HasFailure());` used here is not very elegant.  I guess it's due to that we are doing initialize and tear down in the ctor/dtor of SbPlayerTestFixture, which also prevent us from using ASSERT_NO_FATAL_FAILURE() in the ctor of dtor.

Are we able to turn this into a factory method to return a std::unique_ptr<SbPlayerFixture>, and then use `fixture_.reset();` to clean up, so we can properly use all the GTest macros?

Reply:
I changed it to
```
if (HasFatalFailure()) {
  return;
}
```

From gtest doc (http://go/gunitadvanced#checking-for-failures-in-the-current-test), it seems to be a common case.